cdap-api-spark/src/main/java/co/cask/cdap/api/spark/JavaSparkMain.java

```diff
@@ -55,12 +57,12 @@
  * </pre></code>
  * </p>
  */
-public abstract class JavaSparkMain {
+public interface JavaSparkMain extends Serializable {
```

Comment:
Strictly speaking, this interface doesn't need to be `Serializable`, but it's very common that the user spark program (the one that implements this interface) needs to be serializable (unless you don't use anonymous function but using named static class instead). Marking this interface `Serializable` is for convenience for the user program so that they don't have to `implements Serializable` explicitly.